Lib/functools.py

```diff
@@ -868,3 +868,41 @@ def _method(*args, **kwargs):
     @property
     def __isabstractmethod__(self):
         return getattr(self.func, '__isabstractmethod__', False)
+
+
+################################################################################
+### cached_property() - computed once per instance, cached as attribute
+################################################################################
+
+_NOT_FOUND = object()
+
+
+class cached_property:
+    def __init__(self, func):
+        self.func = func
+        self.attrname = func.__name__
```

Comment:
This looks redundant since `attrname` is set in `__set_name__()`

Reply:
What do you suggest to do instead? This is the most reasonable fallback in case `__set_name__` is not called in some unusual use of this class, and it avoids the object ever being in an incompletely initialized state.